packages/Interface/src/DTK_C_API.cpp

```diff
@@ -311,26 +314,33 @@ const char *DTK_git_commit_hash()
 
 DTK_Handle DTK_create( DTK_ExecutionSpace space )
 {
-    return new DataTransferKit::DTK_Struct( space );
+    auto handle = new DataTransferKit::DTK_Registry( space );
+    DataTransferKit::valid_handles.insert( handle );
+    return handle;
 }
 
-void DTK_destroy( DTK_Handle handle )
+bool DTK_is_valid( DTK_Handle handle )
 {
-    auto dtk = reinterpret_cast<DataTransferKit::DTK_Struct *>( handle );
+    return DataTransferKit::valid_handles.count( handle );
+}
 
-    if ( dtk )
+void DTK_destroy( DTK_Handle handle )
+{
+    if ( DataTransferKit::valid_handles.count( handle ) )
+    {
+        auto dtk = reinterpret_cast<DataTransferKit::DTK_Registry *>( handle );
```

Comment:
I was going to ask you to remind me why you were using `reinterpret_cast` rather than `static_cast` and I ended up reading [this](https://stackoverflow.com/questions/573294/when-to-use-reinterpret-cast).  I am commenting here for reference.